x/trace.go

```diff
@@ -0,0 +1,32 @@
+package x
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+)
+
+func TraceRequest(ctx context.Context, req *http.Request) func() {
+	tracer := opentracing.GlobalTracer()
+	if tracer == nil {
+		return func() {}
+	}
+
+	parentSpan := opentracing.SpanFromContext(ctx)
+	opts := make([]opentracing.StartSpanOption, 0, 1)
+	if parentSpan != nil {
+		opts = append(opts, opentracing.ChildOf(parentSpan.Context()))
+	}
+
+	urlStr := req.URL.String()
+	clientSpan := tracer.StartSpan(req.Method+" "+urlStr, opts...)
```

Comment:
Is this standard procedure for spans to have the URL be part of the psan title?

Reply:
This is just moved from `pipeline/authn/authenticator_oauth2_introspection.go`. I'm sorry I forgot delete original. Fixed at [df9bd84](https://github.com/ory/oathkeeper/pull/931/commits/df9bd841af9adeb3691721a3fe76e5fc3043f72e).
https://github.com/ory/oathkeeper/blob/69ad28f87fd4f49e1ca5ed0d4bd1099b61a795a5/pipeline/authn/authenticator_oauth2_introspection.go#L174-L195

If we can change it, how about fixed string?

* `HTTP Client` ([opentracing-contrib](https://github.com/opentracing-contrib/go-stdlib/blob/fec77c82c616b7808887ad165958a7d55046eaa6/nethttp/client.go#L238-L240))
* `http.client` ([datadog](https://github.com/DataDog/dd-trace-go/blob/24cd31a806a03484f88ceebd67231b492ab7d1f2/contrib/net/http/roundtripper.go#L42))

Reply:
Sorry for the late reply, it looks like this is the correct approach:

https://github.com/opentracing-contrib/go-stdlib/blob/fec77c82c616b7808887ad165958a7d55046eaa6/nethttp/client.go#L245

```
h.sp = h.tr.StartSpan("HTTP "+req.Method, opentracing.ChildOf(ctx), ext.SpanKindRPCClient)
```

Reply:
Thank you! I fixed at [01e0a96](https://github.com/ory/oathkeeper/pull/931/commits/01e0a96138119703864fb79e1c6c7d8b92c80267).